### ISSUE TITLE
uv 0.2.0 no longer working inside conda environments

### ISSUE THREAD

**Issue by notatallshaw-gts** [astral-sh/uv#3769]
First install conda and activate base environment:

```
$ uv -V
uv 0.2.0

$ which python
~/miniforge3/bin/python

$ python -V
Python 3.10.6

$ uv pip install --dry-run db-dtypes==1.0.3
error: No Python default installation found in virtual environments

$ uv pip install --dry-run db-dtypes==1.0.3 -vvv
    0.001455s  INFO uv_interpreter::virtualenv Found active virtual environment (via CONDA_PREFIX) at: /home/STRIKETECH/dshaw/miniforge3
    0.001548s DEBUG uv_interpreter::discovery Ignoring Python interpreter at `/home/STRIKETECH/dshaw/miniforge3/bin/python`: system intepreter not allowed
error: No Python default installation found in virtual environments uv_requirements::specification::from_source source=db-dtypes==1.0.3
    0.001840s DEBUG uv_interpreter::discovery Searching for interpreter that fulfills Python @ default
    0.001883s DEBUG uv_interpreter::discovery Searching for interpreter that fulfills Python @ default
    0.001909s  INFO uv_interpreter::virtualenv Found active virtual environment (via CONDA_PREFIX) at: /home/STRIKETECH/dshaw/miniforge3
    0.002027s DEBUG uv_interpreter::discovery Ignoring Python interpreter at `/home/STRIKETECH/dshaw/miniforge3/bin/python`: system intepreter not allowed
error: No Python default installation found in virtual environments
```

But this is not my system interpreter.

**Comment by zanieb** [comment]
Thanks this looks like a bug in our interpreter identification — `Interpreter::is_virtualenv` is returning false here.

Can you share the output of 

```
/home/STRIKETECH/dshaw/miniforge3/bin/python -c "import sys; print(sys.prefix); print(sys.base_prefix)"
```

**Comment by charliermarsh** [comment]
It's possible that Conda environments don't adhere to that definition.

**Comment by notatallshaw-gts** [comment]
```
$ /home/STRIKETECH/dshaw/miniforge3/bin/python -c "import sys; print(sys.prefix); print(sys.base_prefix)"
/home/STRIKETECH/dshaw/miniforge3
/home/STRIKETECH/dshaw/miniforge3
```

**Comment by charliermarsh** [comment]
They're not actually virtual environments so I think that's reasonable.

**Comment by zanieb** [comment]
Yeah that's fair. It's a little awkward for us when we're trying to ensure that you're working in a environment that's safe to mutate without opt-in.

**Comment by notatallshaw-gts** [comment]
Yeah, conda environments act like system environments that the the user can create and throw away, you can create virtual environments from a conda environment to work around this, but it's a level of redundancy that is a waste for most use cases.

**Comment by notatallshaw-gts** [comment]
> Yeah that's fair. It's a little awkward for us when we're trying to ensure that you're working in a environment that's safe to mutate without opt-in.

Btw, as the first step not to check for PEP 668 externally managed flag?

For any newish Linux distributions this is probably sufficient, older ones probably not 🙁.

**Comment by zanieb** [comment]
Yeah we do check for that, but that's another level of opt-in (matching pip's interface).

I've got a fix in progress at #3771

**Comment by zanieb** [comment]
Release on its way at #3774
